vmm/src/vm.rs

```diff
@@ -1913,6 +1917,20 @@ impl Vm {
             self.vm.tdx_finalize().map_err(Error::FinalizeTdx)?;
         }
 
+        let mode = self.config.lock().unwrap().serial.mode;
+        match mode {
+            ConsoleOutputMode::Pty | ConsoleOutputMode::Tty => {
+                self.serial_handle = Some(
+                    self.device_manager
+                        .lock()
+                        .unwrap()
+                        .start_serial_manager_thread(mode)
```

Comment:
Could this just be part of `add_console_device()` function in `DeviceManager`?

Reply:
I wasn't exactly sure about the timing of when to start the thread. I'll try that out.